apps/playground-web/src/app/connect/auth/page.tsx

```diff
@@ -0,0 +1,178 @@
+              <Button asChild size="lg">
+                <Link
+                  target="_blank"
+                  href="https://portal.thirdweb.com/typescript/v5/auth"
+                >
+                  View docs
+                </Link>
+              </Button>
+              <Button asChild variant="outline" size="lg">
+                <Link target="_blank" href="https://thirdweb.com/contact-us">
+                  Book a Demo
+                </Link>
+              </Button>
+            </div>
+          </div>
+          <div className="w-full mx-auto my-auto sm:w-full order-first lg:order-last relative flex flex-col space-y-2">
+            <div className="max-w-full sm:max-w-[600px]">
+              <Image
+                src={"/auth.png"}
+                width={600}
+                height={400}
+                objectFit={"contain"}
+                alt=""
+                priority={true}
+              />
+            </div>
+          </div>
+        </div>
+      </section>
+
+      <section className="container px-4 md:px-6 space-y-8">
+        <BasicAuth />
+      </section>
+
+      <section className="container px-4 md:px-6 space-y-8">
+        <GatedContent />
+      </section>
+    </main>
+  );
+}
+
+function BasicAuth() {
+  return (
+    <>
+      <div className="space-y-2">
+        <h2 className="text-2xl sm:text-3xl font-semibold tracking-tight">
+          Basic Auth
+        </h2>
+        <p className="max-w-[600px]">
+          Add authentication to your app with a single component.
+        </p>
+      </div>
+
+      <CodeExample
+        preview={<BasicAuthPreview />}
+        code={`"use client";
+
+import {
+  generatePayload,
+  isLoggedIn,
+  login,
+  logout,
+} from "@/app/connect/auth/server/actions/auth";
+import { THIRDWEB_CLIENT } from "@/lib/client";
+import { ConnectButton } from "thirdweb/react";
+
+export function AuthButton() {
+  return (
+    <ConnectButton
+      client={THIRDWEB_CLIENT}
+      auth={{
+        isLoggedIn: async () => {
+          const authResult = await isLoggedIn();
+          if (!authResult) return false;
+          return true;
+        },
+        doLogin: async (params) => await login(params),
+        getLoginPayload: async ({ address }) => generatePayload({ address }),
+        doLogout: async () => await logout(),
+      }}
+    />
+  );
+}
+`}
+        lang="tsx"
+      />
+    </>
+  );
+}
+
+function GatedContent() {
+  return (
+    <>
+      <div className="space-y-2">
+        <h2 className="text-2xl sm:text-3xl font-semibold tracking-tight">
+          Gating content with Auth
+        </h2>
+        <p className="max-w-[600px]">
+          Protect your page with thirdweb Auth. Deliver exclusive content to
+          users who qualify.
+        </p>
+      </div>
+
+      <CodeExample
+        preview={<GatedContentPreview />}
+        code={`import { THIRDWEB_CLIENT } from "@/lib/client";
+import { cookies } from "next/headers";
```

Comment:
You changed the code here? I actually liked it better when you showed the jwt parsing - needs be clear what's going on from reading the snippet